ee/packages/presence/src/Presence.ts

```diff
@@ -39,9 +39,13 @@ export class Presence extends ServiceClass implements IPresence {
 			}
 		});
 
-		this.onEvent('license.module', ({ module, valid }) => {
+		this.onEvent('license.module', async ({ module, valid }) => {
```

Comment:
I believe this partially fixes the problem. Testing it locally, it looks like this event is not being received (even having a valid license). I'm unsure if this is the intended behavior or a problem with the `onModule` function.

Also, if I'm not wrong,  `started()` method solves the problem. https://github.com/RocketChat/Rocket.Chat/blob/develop/ee/packages/presence/src/Presence.ts#L61

The only problem that could happen is, while `started()` is executing AND before it fully finishes, we receive an event on the `watch.instanceStatus` turning the setting to true again. What I'm trying to say is, the execution must be at [this](https://github.com/RocketChat/Rocket.Chat/blob/develop/ee/packages/presence/src/Presence.ts#L239) point when the [`started()`](https://github.com/RocketChat/Rocket.Chat/blob/develop/ee/packages/presence/src/Presence.ts#L63) finishes.

I think @sampaiodiego opinion is really important here.

Reply:
I don't think this is the best solution either..

if the problem is really the method `validateAvailability` being called **before** the start up procedure, we should actually make sure that doesn't happen by adding some kind of control like `started = true`.. or maybe making like `hasLicense = boolean | null` , so if it is still `null` we ignore all events.

Reply:
Thought of another approach: [here](https://github.com/RocketChat/Rocket.Chat/pull/30282/files#diff-4e5b24fc44a40a717ed7e5f19fcbbb5c978f13eac88725313af59ff3c7def3e9R236)
It's basically calling `toggleBroadcast` directly on new connections if the license is active and presence broadcast is disabled